core/src/main/java/org/opensearch/sql/data/model/ExprTimeValue.java

```diff
@@ -37,9 +44,43 @@ public ExprTimeValue(String time) {
     }
   }
 
+  /** Constructor of ExprTimeValue to support custom/OpenSearch date formats in mappings. */
+  public ExprTimeValue(String time, List<DateFormatter> dateFormatters) {
```

Comment:
Seems it's very similar to ExprDateValue constructor. Can we extract some part and reduce redundancy?

Reply:
Yes I did extracted few date formatting stuff inside DateFormatter util class. I think remaining ones are little more specific to the way we parse the date string in that particular class.